src/detect-rfb-secresult.c

```diff
@@ -286,14 +285,13 @@ static int RfbSecresultTestParse01 (void)
  */
 static int RfbSecresultTestParse02 (void)
 {
-    DetectRfbSecresultData *de = NULL;
-    de = DetectRfbSecresultParse("invalidopt");
-    if (de) {
-        DetectRfbSecresultFree(NULL, de);
-        return 0;
-    }
+    DetectRfbSecresultData *de = DetectRfbSecresultParse("invalidopt");
+
+    FAIL_IF_NOT_NULL(de);
+
+    DetectRfbSecresultFree(NULL, de);
```

Comment:
One review I received just the other day and which I believe applies here is that we can remove this `DetectRfbSecresultFree` call. The reasoning behind it is that if it is not null, we won't reach this call, and if we reach it... then it should be null, already...